src/javascript/app/pages/trade/purchase.js

```diff
@@ -45,7 +45,7 @@ const Purchase = (() => {
         const button             = CommonFunctions.getElementById('contract_purchase_button');
 
         const error      = details.error;
-        const show_chart = !error && passthrough.duration <= 10 && passthrough.duration_unit === 't' && (sessionStorage.formname === 'risefall' || sessionStorage.formname === 'higherlower' || sessionStorage.formname === 'asian');
+        const show_chart = !error && passthrough.duration <= 10 && passthrough.duration_unit === 't' && (sessionStorage.formname === 'risefall' || sessionStorage.formname === 'higherlower' || sessionStorage.formname === 'asian' || sessionStorage.formname === 'touchnotouch');
```

Comment:
wouldn't it be easier to do a regex check instead of the long chain of `sessionStorage.formname === 'risefall' || sessionStorage.formname === 'higherlower' || sessionStorage.formname === 'asian' || sessionStorage.formname === 'touchnotouch'`? something like `/^risefall|higherlower|asian|touchnotouch$/.test(sessionStorage.formname)`